waltz-ng/client/data-types/components/data-type-tree.js

```diff
@@ -0,0 +1,67 @@
+import {initialiseData, buildHierarchies} from "../../common";
+import {summerFactory} from "../../common/tally-utils";
+
+
+const bindings = {
+    dataTypes: '<',
+    tallies: '<'
+};
+
+
+const initialState = {
+    dataTypeHierarchy: []
+};
+
+
+const template = require('./data-type-tree.html');
+
+
+function prepareDataTypeTree(dataTypes, tallies) {
+
+    const dataTypesByCode = _.keyBy(dataTypes, 'code');
+
+    const enrichWithDirectCounts = (tallies, keyName) => {
+        _.each(tallies, t => {
+            const dt = dataTypesByCode[t.id];
+            if (dt) dt[keyName] = t.count;
+        });
+    };
+
+    enrichWithDirectCounts(tallies, "dataFlowCount");
```

Comment:
why have hardcoded string here, would keyName param to enrichWithDirectCounts ever change  ?